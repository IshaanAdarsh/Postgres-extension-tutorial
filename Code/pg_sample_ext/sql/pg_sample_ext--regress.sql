-- regression test script for pg_sample_ext (version 1.0.2)

-- Create the extension
CREATE EXTENSION pg_sample_ext;

-- Test case 1: Verify the square function
SELECT square(5) AS result, 25 AS expected_result;

-- Test case 2: Verify the square function with a negative number
SELECT square(-4) AS result, 16 AS expected_result;


-- Test case 3: Verify the person_type composite type
CREATE TABLE my_table (
id serial PRIMARY KEY,
person person_type
);

INSERT INTO my_table (person)
VALUES (('John', 30));

SELECT * FROM my_table;

-- Test case 4: Verify the status_type enumerated type
CREATE TABLE status_table (
id serial PRIMARY KEY,
status status_type
);

INSERT INTO status_table (status)
VALUES ('active');

SELECT * FROM status_table;

-- Test case 5: Verify the positive_integer domain type
CREATE TABLE positive_integer_table (
id serial PRIMARY KEY,
value positive_integer
);

INSERT INTO positive_integer_table (value)
VALUES (42);

SELECT * FROM positive_integer_table;

-- Test case 6: Verify the @* operator
SELECT ROW(2)::my_type @* ROW(3)::my_type AS result;

