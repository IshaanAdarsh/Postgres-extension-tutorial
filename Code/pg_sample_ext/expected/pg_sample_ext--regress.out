-- regression test script for pg_sample_ext
-- Create the extension
CREATE EXTENSION pg_sample_ext;
-- Test case 1: Verify the square function
SELECT square(5) AS result, 25 AS expected_result;
 result | expected_result 
--------+-----------------
     25 |              25
(1 row)

-- Test case 2: Verify the square function with a negative number
SELECT square(-4) AS result, 16 AS expected_result;
 result | expected_result 
--------+-----------------
     16 |              16
(1 row)

